integrations/appboy/test/index.test.js

```diff
@@ -335,37 +335,61 @@ describe('Appboy', function() {
         );
       });
 
-      it('should handle custom traits of acceptable types and excludes nested hashes', function() {
+      it('should handle string traits', function() {
         analytics.identify('userId', {
-          song: "Who's That Chick?",
-          artists: ['David Guetta', 'Rihanna'],
-          details: { nested: 'object' },
-          number: 16,
-          date: 'Tue Apr 25 2017 14:22:48 GMT-0700 (PDT)'
+          song: "Who's That Chick?"
         });
         analytics.called(window.appboy.changeUser, 'userId');
         analytics.called(
           window.appboy.ab.User.prototype.setCustomUserAttribute,
           'song',
           "Who's That Chick?"
         );
-        analytics.called(
-          window.appboy.ab.User.prototype.setCustomUserAttribute,
-          'artists',
-          ['David Guetta', 'Rihanna']
-        );
+      });
+
+      it('should handle number traits', function() {
+        analytics.identify('userId', {
+          number: 16
+        });
+        analytics.called(window.appboy.changeUser, 'userId');
         analytics.called(
           window.appboy.ab.User.prototype.setCustomUserAttribute,
           'number',
           16
         );
+      });
+
+      it('should handle list traits', function() {
+        analytics.identify('userId', {
+          artists: ['David Guetta', 'Rihanna']
+        });
+        analytics.called(window.appboy.changeUser, 'userId');
+      });
+
+      it('should handle date traits', function() {
+        analytics.identify('userId', {
+          date: 'Tue Apr 25 2017 14:22:48 GMT-0700 (PDT)'
+        });
+        analytics.called(window.appboy.changeUser, 'userId');
         analytics.called(
           window.appboy.ab.User.prototype.setCustomUserAttribute,
           'date',
           'Tue Apr 25 2017 14:22:48 GMT-0700 (PDT)'
         );
       });
 
+      it('should exclude nested hash traits', function() {
```

Comment:
Should this test pass? Looks like the [previous test](https://github.com/segmentio/analytics.js-integrations/blob/a4c8a799d8b73abb21e3fd657bb7b51686d3bfc6/integrations/appboy/test/index.test.js#L338-L367) excluded this nested object.

Reply:
This test will pass because there is still a bug floating around in the code that isn't fixed yet. [This line](https://github.com/segmentio/analytics.js-integrations/blob/master/integrations/appboy/lib/index.js#L327) should actually be:
```
typeof value === 'object' && !Array.isArray(value)
```

Reply:
Ahh gotcha, thanks for the context!